tldr

```diff
@@ -252,8 +258,9 @@ fetch_tldr() {
 }
 
 list_pages() {
-    for platform in $(ls "$configdir" ); do
-        for page in $(ls 2>/dev/null "$configdir"/"$platform/") ; do
+    for platform in "$configdir"/* ; do
+        [ -e "$platform" ] || break
+        for page in "${configdir}/${platform}/"* ; do
```

Comment:
This may fail if $configdir doesn't exist, or is empty. Was this tested?

Reply:
The script calls `config` which populates the config folder beforehand, so this works. The problem that I saw when double-checking it that at the top of the script `set -f` is called which disables globbing. Is that option necessary?

Reply:
True. Well, what you're trying to change it -to- is what I started with, and had to change to the ls version. I'll try to remember why I needed the change, whether it was due to POSIX weirdness, the dash shell, or what.

Reply:
You can try running the tests, by the way, to help make sure the changes are okay:

https://github.com/raylee/tldr/blob/master/test

Reply:
Okay, I remember the problem here. It's this issue with shell globbing:

$ set -f; for i in does-not-exist/*; do echo $i; done;
does-not-exist/*
$ set +f; for i in does-not-exist/*; do echo $i; done;
does-not-exist/*

The desired behavior is to have the loop never execute, rather than blindly echoing the loop's parameters.